maintenance/monitor-component-metrics.md

```diff
@@ -3,43 +3,261 @@ title: Monitor Calico component metrics
+      external_labels:
+        monitor: 'tutorial-monitor'
+    scrape_configs:
+    - job_name: 'prometheus'
+      scrape_interval: 5s
+      static_configs:
+      - targets: ['localhost:9090']
+    - job_name: 'felix_metrics'
+      scrape_interval: 5s
+      scheme: http
+      kubernetes_sd_configs:
+      - role: endpoints
+      relabel_configs:
+      - source_labels: [__meta_kubernetes_service_name]
+        regex: felix-demo-svc
+        replacement: $1
+        action: keep
+    - job_name: 'typha_metrics'
+      scrape_interval: 5s
+      scheme: http
+      kubernetes_sd_configs:
+      - role: endpoints
+      relabel_configs:
+      - source_labels: [__meta_kubernetes_service_name]
+        regex: typha-demo-svc
+        replacement: $1
+        action: keep
+EOF
+```
 
-#### Enable Prometheus metrics for and Felix and Typha
+## 3. Creating a service to expose Felix metrics
 
-1. Using the Prometheus documentation, configure one or more [Prometheus servers](https://prometheus.io/docs/introduction/overview/).  
-1. To enable [Felix]({{ site.baseurl }}/reference/felix/configuration) for metrics, set **PrometheusMetricsEnabled = true**.
-1. To enable [Typha]({{ site.baseurl }}/reference/typha/configuration) for metrics, set **PrometheusMetricsEnabled = true**.
-1. If required for Felix and/or Typha, change the default TCP port (9091) for your Prometheus metrics server using the parameter, **PrometheusMetricsPort**.
+Your prometheus configuration is going to check services and catch endpoints related to felix-demo-svc service, you need to create a service with that name to expose Felix metrics. This will help in discovery of your services and **makes** it dynamic. 
 
-#### Best practices
+> **Note**: Felix uses port 9091 TCP to publish its metrics.
+   {: .alert .alert-info}
+
+``` bash
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Service
+metadata:
+  name: felix-demo-svc
+  namespace: kube-system
+spec:
+  selector:
+    k8s-app: calico-node
+  ports:
+  - port: 9091
+    targetPort: 9091
+EOF
+```
+
+## 4. Creating a service to expose Typha metrics
+
+You need to create a service that exposes your typha instance metrics to Prometheus.
+
+> **Note**: Typha uses **port 9091** TCP to publish its metrics. However, in AWS Implementation of calico port is set to TCP Port 9093 via **TYPHA_PROMETHEUSMETRICSPORT** environment variable.
+   {: .alert .alert-warning}
+
+``` bash
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Service
+metadata:
+  name: typha-demo-svc
+  namespace: kube-system
+spec:
+  selector:
+    k8s-app: calico-typha
+  ports:
+  - port: 9093
+    targetPort: 9093
+EOF
+```
+
+## 5. Creating Prometheus pod
+
+Now that you allowed the user to view the metrics and have a valid config file for your Prometheus, it's time to create the pod.
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Pod
+metadata:
+  name: prometheus-pod
+  labels:
+    app: prometheus-pod
+    role: monitoring
+spec:
```

Comment:
I was kind of expecting to see the service account we created earlier to be included in this pod spec.  Do you know how prometheus ends up with that service account if it isn't set here?

Reply:
To the best of my knowledge, Prometheus in this setup will try to gather information via "system:serviceaccount:default:default" credentials.

Assuming our default context when following this tutorial is **default**, Our pod will be assigned to **default** namespace since we have not specified any namespace for it. this will create the following
`system:serviceaccount:default`
As you have mentioned because we have [not specified any service accounts](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#use-the-default-service-account-to-access-the-api-server) for our pod it will be "default" and we will end up with.
`system:serviceaccount:default:**default**`